proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+                        "fieldNumber": 1
+                    },
+                    "nftID": {
+                        "dataType": "bytes",
+                        "length": NFT_ID_LENGTH,
+                        "fieldNumber": 2
+                    },
+                    "escrow": {
+                        "dataType": "boolean",
+                        "fieldNumber": 3
+                    }
+                }
+            }
+        },
+        "supportedNFTsSubstore": {
+            "type": "array",
+            "fieldNumber": 4,
+            "items": {
+                "type": "object",
+                "required": [
+                    "chainID",
+                    "supportedNftIDArray"
+                ],
+                "properties": {
+                    "chainID": {
+                        "dataType": "bytes",
+                        "length": CHAIN_ID_LENGTH,
+                        "fieldNumber": 1
+                    },
+                    "supportedNftIDArray": {
+                        "type": "array",
+                        "fieldNumber": 2,
+                        "items": {
+                            "dataType": "bytes",
+                            "length": NFT_ID_LENGTH
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
 ```
 
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the NFT module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisNFTStoreSchema` schema, given above.
+
+- Initial checks on the properties of `genesisBlockAssetObject`:
+
+  - Across all elements of the `NFTSubstore` array, all values given for `nftID` must be unique.
+  - For all elements of the `NFTSubstore` array, values given for `owner` must have either length `ADDRESS_ID_LENGTH` bytes (representing a user address) or `CHAIN_ID_LENGTH` bytes (representing a chain ID).
+  - The `NFTSubstore` must be in lexicographical order of `owner`.
+    - For a given `owner`, the entries must be in ascending order of `nftID`.
+    - The `attributesArray` must be in lexicographic order of `module` name.
+  - Across all elements of the `userSubstore` array, the pairs `address`, `nftID` must be unique: each pair appears at most once, although a given `address` can appear multiple times with different unique `nftID`.
+  - The `userSubstore` array must be in lexicographical order of `address`. For a given `address`, the entries must be in lexicographic order of `nftID`.
+  - Across all elements of the `escrowSubstore` array, the pairs `escrowedChainID`, `nftID` must be unique: each pair appears at most once, although a given `escrowedChainID` can appear multiple times with different unique `nftID`.
```

Comment:
same as above, `nftID` should be independently unique

Reply:
Just to confirm: we do not check that all entries in NFT substore have a corresponding entry in either user or escrow substore because in principle the NFT could have been destroyed?

Even in this case, I think we should check that all entries in user and escrow substore together must have a corresponding entry in the NFT substore.